rbi/stdlib/fileutils.rbi

```diff
@@ -257,7 +257,7 @@ module FileUtils
     .params(
       dir: T.any(String, Pathname),
       verbose: T.nilable(T::Boolean),
-      block: T.proc.returns(T.type_parameter(:U)),
+      block: T.nilable(T.proc.returns(T.type_parameter(:U))),
     )
     .returns(T.type_parameter(:U))
```

Comment:
In the case where no block is passed, Sorbet is saying that since it didn't see any input given of type `T.type_parameter(:U)`, it will be impossible for the method to return anything. So assuming that the implementation is well-typed according to this signature, the function must raise.

To properly type this, you will have to overload the signature, or change the return type to `T.any(Integer, T.type_parameter(:U))`, because it looks like this method returns `0` if no block was passed.

Sorbet doesn't support using overloaded signatures with methods defined using `module_function`, so if you want to overload this signature, you will also have to duplicate the `self.` and non-`self.` definitions of the method.

Reply:
Ah, ok, i think it's easier at this point to just pass an empty block in my code. Thanks for elaborating!

Reply:
I think if you pass an empty block, when the block finishes, Ruby will cd back to the original folder before the cd, effectively cancelling itself out.

Reply:
ah, yep, 🤦, but i still have an easy workaround. thanks!